### PR TITLE
Website selector: searching for special strings will show html code

### LINKED ISSUE #7692: Website selector: searching for special strings will show html code
Reproduce:
 * in a website selector, with 10 websites or so,
 * enter in the search field `an` (or any substring of `...<span class="autocompleteMatched">....</span>....
 * notice that the autocompleter will highlight in purple some text but it actually displays html code of the element, as can be seen in this screenshot:

highlighter

Expected behavior would be that the html code is not displayed (and that the highlighter / search does not match on the html text but only on the text without markup)

### PR DESCRIPTION
**Pull request by tsteur**
fixes #7692

This worked for me. Can someone please try as well? Sometimes I can still see something like `tocomplete` for a second but it should be shown correctly immediately after.

### CODE REVIEW

File: plugins/CoreHome/angularjs/common/directives/autocomplete-matched.js [hunk 1]
```diff
@@ -19,26 +19,30 @@
...
     function piwikAutocompleteMatched() {
-        return function(scope, element, attrs) {
-            var searchTerm;
 
-            scope.$watch(attrs.piwikAutocompleteMatched, function(value) {
-                searchTerm = value;
-                updateText();
-            });
+        return {
+            priority: 10,
```
Comment (tsteur) [inline plugins/CoreHome/angularjs/common/directives/autocomplete-matched.js]: Basically this is the fix. We tell the directive to render after other directives

**Comment by mnapoli** [comment]
Can still see HTML code appear when typing.

Also there is this kind of encoding issue:

capture d ecran 2015-07-30 a 15 03 22

Whereas the name of the site is:

capture d ecran 2015-07-30 a 15 04 12

Other example:

capture d ecran 2015-07-30 a 15 06 19

**Comment by mnapoli** [comment]
Also given the obscure nature of the fix, it would be great to document it (unless another solution is applied given it doesn't solve all the problems).

**Comment by tsteur** [comment]
As mentioned it can be still visible for like half a second or so but should be quickly displayed correct.  Obscure nature of the fix? Do you mean the priority? Priorities are a common thing in angular and not really obscure I'd say. Can leave a comment though. 

Will have a look re the problem if the search term is eg `lt`

**Comment by tsteur** [comment]
Pushed another commit

**Comment by mnapoli** [comment]
> Do you mean the priority? Priorities are a common thing in angular and not really obscure I'd say. Can leave a comment though.

```js
priority: 10, // makes sure to render after other directives
```

The concept of priority is obvious, what's less obvious is why it's important to render after the other directives. That's what won't be obvious to someone landing on that code in a few weeks/months/years.

**Comment by tsteur** [comment]
I'm not sure what to comment there. It should simply run last otherwise it could be overwritten again by another directive. Eg in my case the colors / autocomplete match was often not visible. That's why there's a priority set indicating it should run after other directives. I can link to this PR if it helps

**Comment by mattab** [comment]
Well done, looks fixed!

**Pull request merged**